packages/app/tasks/get-pool-pending-rewards.js

```diff
@@ -0,0 +1,318 @@
+                                    });
+                                }
+                            }
+
+                            if (matches.length !== 1) {
+                                error(
+                                    'Failed to fully match pool and reserve tokens. Expected to find a single match, but found',
+                                    arg('matches', matches.length)
+                                );
+                            }
+                        } else if (matches.length !== 1) {
+                            error(
+                                'Failed to fully match pool and reserve tokens. Expected to find a single match, but found',
+                                arg('matches', matches.length)
+                            );
+                        }
+
+                        const { poolToken, reserveToken } = matches[0];
+
+                        if (!eq(poolToken, targetPoolToken)) {
+                            continue;
+                        }
+
+                        set(poolData, [provider, poolToken, reserveToken], {});
+
+                        eventCount++;
+
+                        data.lastBlockNumber = blockNumber;
+
+                        break;
+                    }
+
+                    case 'ProtectionRemoved': {
+                        const provider = returnValues._provider;
+                        const poolToken = returnValues._poolToken;
+                        const reserveToken = returnValues._reserveToken;
+
+                        trace(
+                            'Found ProtectionRemoved event at block',
+                            arg('blockNumber', blockNumber),
+                            arg('provider', provider),
+                            arg('poolToken', poolToken),
+                            arg('timestamp', timestamp),
+                            arg('tx', transactionHash)
+                        );
+
+                        if (!eq(poolToken, targetPoolToken)) {
+                            continue;
+                        }
+
+                        set(poolData, [provider, poolToken, reserveToken], {});
+
+                        eventCount++;
+
+                        data.lastBlockNumber = blockNumber;
+
+                        break;
+                    }
+                }
+            }
+        }
+
+        info('Finished processing all new protection change events', arg('count', eventCount));
+
+        return poolData;
+    };
+
+    const getPendingRewards = async (data, poolData, toBlock) => {
+        info('Querying all pending rewards at block', arg('toBlock', toBlock));
+
+        const { pendingRewards = {} } = data;
+
+        for (const [provider, poolTokens] of Object.entries(poolData)) {
+            for (const [poolToken, reserveTokens] of Object.entries(poolTokens)) {
+                for (const reserveToken of Object.keys(reserveTokens)) {
```

Comment:
Minor: can use `for (const reserveToken in reserveTokens)`.

Reply:
Oohhh, nice! I completely forgot about this trick!

Reply:
The common problem here is not to forget about it so much as to remember it, because once you start using `for...in` too often, the minute you apply it in the wrong place (anything other than the purpose of iterating the keys of an object) - you're in a world of hurt.

Reply:
LOL. I can definitely see it happening...